### PR TITLE
[fml] create shared formulas

### PR DESCRIPTION
**Pull request by JanMarvin**
``` r
library(openxlsx2)

df <- data.frame(
  x = 1:5,
  y = 1:5 * 2
)

wb <-  wb_workbook()$add_worksheet()$add_data(x = df)

wb$add_formula(
  x      = "=A2/B2",
  dims   = "C2:C6",
  shared = TRUE
)

wb$add_formula(
  x      = "=A$2/B$2",
  dims   = "D2:D6",
  shared = TRUE
)

if (interactive()) wb$open()
```

### CODE REVIEW

**Comment by JanMarvin** [comment]
My inspiration was this [SO](https://stackoverflow.com/a/78718986/12340029). Maybe you can double check @olivroy ? Just to make sure that nothing unintended happens.

This is somewhat related to #1072

**Comment by olivroy** [comment]
I can check more thoroughly, but may be worth adding an example of this somewhere?

File: R/write.R [hunk 1]
```diff
@@ -459,6 +461,21 @@ write_data2 <- function(
...
+    # This cc contains only the formula range.
+    ## the top left cell is the reference
+    ## all have shared and all share the same f_si
+    ## only the reference cell has a formula
+    ## only the reference cell has the formula reference
+    cc$f_t              <- "shared"
+    cc[1, "f_ref"]      <- dims
+    cc[2:nrow(cc), "f"] <- ""
+
+    int_si <- as.integer(wb$worksheets[[sheetno]]$sheet_data$cc$f_si)
```
Comment (JanMarvin) [inline R/write.R:474]: Use unique

**Comment by JanMarvin** [comment]
You always think in the bigger picture 👍  

😄 My line of thought was something like this
* hm this SO does not work
* hm I can make this work
* hm maybe this can be solved with shared strings
* oh my tiny PR looks like it's working
* darn, does not work
* why doesn't it work!
* ah, if I exclude this, it might work
* what is this `enforce` argument and why isn't it documented?
* gnar I have to fix this
* oh it works!
* Lets push to github, maybe Oliver knows what kind of checks are still missing

**Comment by olivroy** [comment]
Don't forget to increment version number too!

**Pull request merged**